### ISSUE TITLE
Style all scrollbars in Firefox as they are in Chrome

### ISSUE THREAD

**Issue by facelessuser** [squidfunk/mkdocs-material#1717]
__I checked that...__

- [X] ... the documentation does not mention anything about my idea
- [X] ... to my best knowledge, my idea wouldn't break something for other users
- [X] ... there are no open or closed issues that are related to my idea

## Description

Basically we need the equivalent of (obviously just place in the correct places):

```css
.md-typeset__scrollwrap,
.md-sidebar__scrollwrap,
.md-search__scrollwrap,
.md-typeset pre > code {
  scrollbar-color: var(--md-default-fg-color--lighter) transparent;
  scrollbar-width: thin;

  &:hover {
    scrollbar-color: var(--md-accent-fg-color) transparent;
  }
}
```

This also fixes giant scrollbars in table scroll wraps by styling `.md-typeset__scrollwrap`.

I plan on creating a pull request unless someone gets to it before me.

### Use Cases

Create consistent scroll bars across browsers.

### Screenshots / Mockups

Basically it looks like Chrome scrollbars, but in FireFox. Don't really need a picture. Reference #1686.

**Comment by squidfunk** [comment]
On iOS (and other browsers that support "invisible" scroll bars), I'd really like to leave the table scroll wrappers styled as they are. The overflow is clearly visible and encourages scrolling:

<img width="379" alt="Bildschirmfoto 2020-05-22 um 10 33 23" src="https://user-images.githubusercontent.com/932156/82648300-c61a1380-9c17-11ea-83df-7cfce130becf.png">

**Comment by facelessuser** [comment]
Hmm, can you just make the scroll bars invisible on all platforms by choosing transparent for the color on tables?

**Comment by facelessuser** [comment]
Never mind, I get what you mean. Just tried it out on Android. Invisible until you scroll.

**Comment by facelessuser** [comment]
I'll just leave the "invisible" scroll bars alone. The rest can still be styled. I wasn't aware of a reason to not style them until now :slightly_smiling_face:.

**Comment by squidfunk** [comment]
It would be interesting to know when there are ugly scroll bars on tables. Windows-only? What about Linux? AFAIK, scrollbars are OS-dependent.

**Comment by facelessuser** [comment]
I know it happens in Windows, but my mac is so old right now, I mainly use my Windows work laptop at home, and since the pandemic hit, I'm not on my Linux box in the office either. It's not a big deal anyways as I rarely see table scrollbars on a desktop. It was more a thing I noticed when patching scrollbars and forcing them to verify the change. But yeah, it is most likely an OS thing.

**Comment by squidfunk** [comment]
Hmm, doesn't work on OSX. I see no difference in Firefox.

**Comment by squidfunk** [comment]
Ah okay, cool. Can't test. No windows 😅

**Comment by facelessuser** [comment]
Ugh, why does mac have to do things so different. Yeah, mac seems to override with there own overlay scrollbars...I'm not sure if that can be turned off or not. There's got to be a way, but I'd have to do some research. Maybe there isn't anything that can be done on mac, but I'm pretty sure it works in Linux too, as I've used this technique in Linux as well.

**Comment by facelessuser** [comment]
If there is no way in Firefox on macos, maybe we just ignore that difference.

**Comment by squidfunk** [comment]
Yes, we can ignore Firefox on macOS. That makes it impossible for me to test, though. Also note that the `:hover` state is set on `::-webkit-scrollbar-thumb` and not on the overflowing container, but I guess we can model it like this in Firefox.

**Comment by wilhelmer** [comment]
At least consider `scrollbar-width: thin` for `.md-sidebar__scrollwrap`, I added that months ago to my custom CSS. It's super nice for FF Windows and I haven't noticed any downsides in other browsers.

**Comment by squidfunk** [comment]
Sure, never said we won't consider it. I'm waiting for @facelessuser's PR 😊

**Comment by facelessuser** [comment]
I'll try to get the pull up sometime soon.

**Comment by facelessuser** [comment]
> Also note that the :hover state is set on ::-webkit-scrollbar-thumb and not on the overflowing container, but I guess we can model it like this in Firefox.

Yeah, I was aware of this as well. It was the closest we could get.
